src/AnalysisPrograms/Recognizers/PteropusSpecies.cs

```diff
@@ -0,0 +1,450 @@
+            // No operation - do nothing. Feel free to add your own logic.
+            base.SummariseResults(settings, inputFileSegment, events, indices, spectralIndices, results);
+        }
+
+        /// <summary>
+        /// This method is called once per segment (typically one-minute segments).
+        /// </summary>
+        /// <param name="audioRecording">one minute of audio recording.</param>
+        /// <param name="genericConfig">config file that contains parameters used by all profiles.</param>
+        /// <param name="segmentStartOffset">when recording starts.</param>
+        /// <param name="getSpectralIndexes">not sure what this is.</param>
+        /// <param name="outputDirectory">where the recogniser results can be found.</param>
+        /// <param name="imageWidth"> assuming ????.</param>
+        /// <returns>recogniser results.</returns>
+        public override RecognizerResults Recognize(AudioRecording audioRecording, Config genericConfig, TimeSpan segmentStartOffset, Lazy<IndexCalculateResult[]> getSpectralIndexes, DirectoryInfo outputDirectory, int? imageWidth)
+        {
+            // Examples of the APIs available. You don't need all of these commands! Pick and choose.
+            if (ConfigFile.HasProfiles(genericConfig))
+            {
+                string[] profileNames = ConfigFile.GetProfileNames<Config>(genericConfig);
+                int count = profileNames.Length;
+                var message = new StringBuilder("Found " + count + " config profile(s): ");
+                foreach (string s in profileNames)
+                {
+                    message.Append(s + ", ");
+                }
+
+                log.Info(message.ToString());
+            }
+            else
+            {
+                log.Warn("No configuration profiles found. Three profiles expected for the Flying Fox recogniser.");
+            }
+
+            RecognizerResults results1 = null;
+
+            if (ConfigFile.TryGetProfile(genericConfig, "Territorial", out var profile1))
+            {
+                results1 = TerritorialCall(audioRecording, genericConfig, profile1, outputDirectory, segmentStartOffset);
+                log.Info("Territory event count = " + results1.Events.Count);
+            }
+            else
+            {
+                log.Warn("Could not access Territorial configuration parameters");
+            }
+
+            RecognizerResults results2 = null;
+            if (ConfigFile.TryGetProfile(genericConfig, "Wingbeats", out var profile2))
+            {
+                results2 = WingBeats(audioRecording, genericConfig, profile2, outputDirectory, segmentStartOffset);
+                log.Info("Wingbeat event count = " + results2.Events.Count);
+            }
+            else
+            {
+                log.Warn("Could not access Wingbeats configuration parameters");
+            }
+
+            // combine the results
+            if (results1 != null && results2 != null)
```

Comment:
this logic is flawed. If either has results the neither will be returned.

I strongly suggest changing the `Wingbeats` and `TerritorialCall` methods so that they do not return `null` and instead return an object with empty arrays. That makes it much easier to concatenate results.

e.g. `results1.Events.Concat(results2.Events.Concat)`